pkgs/development/python-modules/plotpy/default.nix

```
@@ -0,0 +1,53 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  cython_0,
+  numpy_2,
+  setuptools,
+  guidata,
+  pillow,
+  qwt,
+  scikit-image,
+  scipy,
+  tifffile,
+}:
+
+buildPythonPackage rec {
+  pname = "plotpy";
+  version = "2.6.2";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "PlotPyStack";
+    repo = "PlotPy";
+    rev = "v${version}";
+    hash = "sha256-/TV5JfnrG/lxtoUFoHjxL7+iw1/x1gVlfohS3tUipRk=";
+  };
+
+  nativeBuildInputs = [
+    cython_0
+    setuptools
+  ];
+
+  propagatedBuildInputs = [
+    guidata
+    # Doesn't support numpy_1
+    numpy_2
```

```suggestion
    numpy
```
Add `meta.broken` instead.

---

> Add `meta.broken` instead.

So then Hydra will never build it, and a user who would want to use plotpy will have to rebuild all of the following packages with `numpy_2` instead of `numpy_1`:

1. `scipy`
2. `pythran`
3. `h5py`
4. `contourpy`

Does that really makes sense to you? When all of the above packages explicitly state that they support both numpy versions in runtime?

---

That's why I suggest adding a top-level package with the appropriate overrides. Call it plotpy-test if you want.

---

> That's why I suggest adding a top-level package with the appropriate overrides. Call it plotpy-test if you want.

That's so weird and confusing - what would you write in the description of it? How would people understand why are there two varients of `plotpy`? Would that package propagate `numpy_2`, and `scipy` that is built against `numpy_2` etc? Why would you add it the `-test` suffix if someone can really use it in their Python development environment?

---

It's not another version of plotpy. Just one that creates a Python environment with overridden dependencies.